src/Pathfinder.jl

```diff
@@ -24,15 +24,15 @@ export PathfinderResult, MultiPathfinderResult
 export pathfinder, multipathfinder
 
 const DEFAULT_HISTORY_LENGTH = 6
-const DEFAULT_LINE_SEARCH = LineSearches.HagerZhang()
-const DEFAULT_LINE_SEARCH_INIT = LineSearches.InitialHagerZhang()
+const DEFAULT_LINE_SEARCH_CONSTRUCTOR = LineSearches.HagerZhang
+const DEFAULT_LINE_SEARCH_INIT_CONSTRUCTOR = LineSearches.InitialHagerZhang
```

Comment:
Let's keep the constant name and just make it the constructor instead of the object.
```suggestion
const DEFAULT_LINE_SEARCH = LineSearches.HagerZhang
const DEFAULT_LINE_SEARCH_INIT = LineSearches.InitialHagerZhang
```